spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -400,6 +401,31 @@ class DeltaAnalysis(session: SparkSession)
           throw DeltaErrors.notADeltaTableException("RESTORE")
       }
 
+    // Resolve as a resolved table if the path is for delta table. For non delta table, we keep the
+    // path and pass it along. This is needed as DESCRIBE DETAIL command currently supports both
+    // delta and non delta path.
+    case u: UnresolvedPathBasedTable =>
+      val table = getPathBasedDeltaTable(u.path)
+      val tableExists = try {
+        table.tableExists
+      } catch {
+        case NonFatal(e) => false
+      }
+      if (!tableExists) {
+        u
+      } else {
+        val catalogName: Option[String] = table.catalogTable match {
+          case Some(ct) if ct.identifier.catalog.isDefined => Some(ct.identifier.catalog.get)
```

Comment:
I think we should get the catalog name from the identifier instead of table.catalogTable

Reply:
https://github.com/databricks/runtime/pull/70245#discussion_r1320865425